src/DropdownToggle.js

```diff
@@ -7,80 +7,82 @@ import BaseDropdownToggle from 'react-overlays/DropdownToggle';
 import React from 'react';
 
 import Button from './Button';
-import { createBootstrapComponent } from './ThemeProvider';
+import { useBootstrapPrefix } from './ThemeProvider';
 
 const wrapRef = props => {
   const { ref } = props;
   props.ref = ref.__wrapped || (ref.__wrapped = r => ref(findDOMNode(r)));
   return props;
 };
 
-class DropdownToggle extends React.Component {
-  static propTypes = {
-    /**
-     * @default 'dropdown-toggle'
-     */
-    bsPrefix: PropTypes.string,
+const propTypes = {
+  /**
+   * @default 'dropdown-toggle'
+   */
+  bsPrefix: PropTypes.string,
 
-    /**
-     * An html id attribute, necessary for assistive technologies, such as screen readers.
-     * @type {string|number}
-     * @required
-     */
-    id: isRequiredForA11y(PropTypes.any),
+  /**
+   * An html id attribute, necessary for assistive technologies, such as screen readers.
+   * @type {string|number}
+   * @required
+   */
+  id: isRequiredForA11y(PropTypes.any),
 
-    split: PropTypes.bool,
+  split: PropTypes.bool,
 
-    as: PropTypes.elementType,
+  as: PropTypes.elementType,
 
-    /**
-     * to passthrough to the underlying button or whatever from DropdownButton
-     * @private
-     */
-    childBsPrefix: PropTypes.string,
-  };
-
-  static defaultProps = {
-    as: Button,
-  };
+  /**
+   * to passthrough to the underlying button or whatever from DropdownButton
+   * @private
+   */
+  childBsPrefix: PropTypes.string,
+};
 
-  render() {
-    const {
-      bsPrefix,
-      split,
-      className,
-      children,
-      childBsPrefix,
-      as: Component,
-      ...props
-    } = this.props;
+const defaultProps = {
+  as: Button,
+};
 
-    if (childBsPrefix !== undefined) {
-      props.bsPrefix = childBsPrefix;
-    }
+const DropdownToggle = React.forwardRef((props, ref) => {
+  const {
```

Comment:
can you spread this out in the function signature, and name the rest param `props` instead of `otherProps`?

Reply:
Sure @taion 
Can you also help me with adding the ref to the`BaseDropdownToggle` which is a functional component?
Should I pass it directly to the children?

Reply:
i think we need #4661, then, sorry